experimental/traffic-portal/src/app/core/certs/cert-detail/cert-detail.component.ts

```diff
@@ -0,0 +1,134 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { Component, Input, OnChanges } from "@angular/core";
+import { AbstractControl, FormControl, ValidationErrors, ValidatorFn } from "@angular/forms";
+import * as forge from "node-forge";
+import { Hex } from "node-forge";
+
+import { oidToName, pkiCertToSHA1, pkiCertToSHA256 } from "src/app/core/certs/cert.util";
+
+/**
+ * Author contains the information about an author from a cert issuer/subject
+ */
+export interface Author {
+	countryName?: string | undefined;
+	stateOrProvince?: string | undefined;
+	localityName?: string | undefined;
+	orgName?: string | undefined;
+	orgUnit?: string | undefined;
+	commonName: string;
+}
+
+/**
+ * Angular validator that checks if the value is either before or after the given time
+ *
+ * @param before If this is only valid before
+ * @param now The time to compare against
+ * @returns Validator Function
+ */
+function createDateValidator(before: boolean, now: Date): ValidatorFn {
+	return (control: AbstractControl): ValidationErrors | null => {
+		const value = control.value;
+		if (!value) {
+			return null;
+		}
+
+		let valid = false;
+		const d = new Date(value);
+		if (before) {
+			valid = now >= d;
+		} else {
+			valid = now <= d;
+		}
+
+		return valid ? null : {notValid: true};
```

Comment:
`notValid` is a pretty vague validation error. Any validation error at all would make a control "not valid", which is why others use e.g. `required`, `max-length` etc.